editor/components/block-list/block.js

```diff
@@ -286,10 +290,11 @@ export class BlockListBlock extends Component {
 	 * @see https://developer.mozilla.org/en-US/docs/Web/Events/mouseenter
 	 */
 	maybeHover() {
-		const { isMultiSelected } = this.props;
+		const { isMultiSelected, isSelected } = this.props;
 		const { isHovered } = this.state;
 
-		if ( isHovered || isMultiSelected || this.hadTouchStart ) {
+		if ( isHovered || isMultiSelected || isSelected ||
```

Comment:
I explicitly removed `isSelected` here on a previous PR. Let me try to find the reason for this.

Reply:
Alternatively, to mimic the reducer behavior removed in #5078, we could hide the hover effects when the selected block changes the same way we're doing `isTyping` checks in this branch.

Reply:
I think it's not problematic anymore since #5050

Reply:
I didn't really notice any bugs here aside from hitting a debugger breakpoint when I wasn't expecting it. The multi-selection one had some small issue previously where ending a multi-selection would cause hover effects on the starting and ending blocks.

Reply:
>Alternatively, to mimic the reducer behavior removed in #5078, we could hide the hover effects when the selected block changes the same way we're doing `isTyping` checks in this branch.

We might want this anyways. Noting another small issue is that when splitting a block while currently hovered over the original, hover effects appear above the original.